server/routes/api/cards.js

```diff
@@ -1,76 +1,118 @@
 const express = require("express");
 const router = express.Router();
+
+const auth = require("../../middleware/authenticator");
+const { validate, cardValidationRules } = require("../../middleware/validator");
+
 const Card = require("../../models/cards");
 const Column = require("../../models/columns");
 
 // @route POST api/cards/create
 // @desc Create Card
-// @access Public
-router.post("/create", (req, res) => {
-    const name = req.body.name;
-    const desc = req.body.desc;
+// @access Private
+router.post(
+  "/create/:columnId",
+  [auth, cardValidationRules(), validate],
+  async (req, res) => {
+    try {
+      const { name } = req.body;
+      const column = await Column.findById(req.params.columnId);
+      const newCard = { name: name, column: column._id };
 
-    const newCard = { name: name, desc: desc }
-    Card.create(newCard, (err, card) => {
+      Card.create(newCard, async (err, card) => {
         if (err) {
-            console.log(err)
+          console.log(err);
+          res.send(err);
         } else {
-            console.log(card)
-            res.send(card)
+          // Pushing the card to the column
+          column.cards.push({ _id: card.id });
+          await column.save();
+
+          console.log(card);
+          res.send(card);
         }
-    })
-})
-
-// router.put("/update/:id", (req, res) => {
-//     Card.findByIdAndUpdate(req.params.id, req.body, (err, updatedCard) => {
-//         if (err) {
-//             console.log(err)
-//             res.send(err)
-//         } else {
-//             console.log(updatedCard)
-//             res.send(updatedCard)
-//         }
-//     })
-// })
+      });
+    } catch (err) {
+      if (err.name === "CastError")
+        return res.status(404).json({ msg: "Column not found" });
+      res.status(500).send("Server Error");
+    }
+  }
+);
+
+// @route PUT /api/cards/detail/update/:id
+// @desc Move Card to another column
+// @access private
+
+router.put("/detail/update/:id", auth, (req, res) => {
+  try {
+    console.log(req.body);
+    Card.findByIdAndUpdate(req.params.id, req.body, (err, updatedCard) => {
+      if (!updatedCard) return res.status(400).send({ msg: "Invalid Card" });
+
+      console.log("updatedCard", updatedCard);
+      res.send(updatedCard);
+    });
+  } catch (err) {
+    res.status(500).send("Server Error");
+  }
+});
```

Comment:
Curious about where you're planning on calling this route. Is it going to be invoked from the frontend?

Reply:
This endpoint is meant for detail edit card dialog where we edit the desc and save